### DISCUSSION TITLE
Issue with UEFI, unable to reach target in chainload

### DISCUSSION THREAD

**Discussion by AkashPatel2001** [ipxe/ipxe#1050] [General]
I am testing my adapter for iPXE. For legacy mode, it is able to reach the OS installation page; however, UEFI is failing. I cannot blame my driver because I have used the same one to download the iPXE image.

Initial observations:

1) Able to load iPXE image from server.
2) DHCP or ifconfig errors out. [No configuration method succeeded. "Error: Connection timed out"]

**Comment by NiKiZe** [comment]
Moved to discussion.
Provide more details, full error message (including ipxe.org url)
Info about your NIC, and iPXE ifstat output.

**Comment by AkashPatel2001** [comment]
![Screenshot (4)](https://github.com/ipxe/ipxe/assets/140200561/c4c39320-4b89-4d97-913d-745b4e618741)

**Reply by NiKiZe** [reply]
What you show here is the EFI NII interface being used, it shows dhcp being configured followed by a http request that times out. Suggestions and details in the ipxe.org url shown in the message.

What you could do here is to get a iPXE shell, and then run ifstat before and after the failed command.

If you have developed a driver for your NIC you should know how the UNDI, SNP and NII interfaces work.

**Comment by AkashPatel2001** [comment]
Okay, Thanks that would be help. and sorry for #1051 it was mistake. No basically i have not worked before on our nic driver currently in looking in it only.

**Comment by danmcd** [comment]
The following is from the release notes for Chelsio's firmware...

```
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
ERRATA (Open Issues)
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

- 42108: Link timeout takes more than 5 minutes when peer side link is 
         administratively down. 

- 40457: Need support for iPXE.
<rest snipped>
```

Are you, @AkashPatel2001 from Chelsio?  If so I can help you test any fixes you have.  We clearly need some updates somewhere.

**Reply by AkashPatel2001** [reply]
Will soon release driver with fix.

**Comment by danmcd** [comment]
I think https://github.com/illumos/illumos-gate/commit/10816756818b2dfdee26d91c8b748243d5835882 fixed this.

**Comment by danmcd** [comment]
Aka the appropriate chelsio firmware.
